File: RemoteControl/RemoteControl.Rest/src/RemoteControl.Rest.Persistence/Database/Models/Device.cs
using System.ComponentModel.DataAnnotations;

namespace RemoteControl.Rest.Persistence.Database.Models;

/// <summary>
///     Represents a device in the system.
///     Each device may be associated with multiple scripts through the
///     <see cref="DeviceScriptsMappings" />.
/// </summary>
public class Device
{
    /// <summary>
    ///     Gets or sets the unique identifier for the device.
    /// </summary>
    public required int Id { get; set; }

    /// <summary>
    ///     Gets or sets the name of the device.
    /// </summary>
    [MaxLength(255)]
    public required string Name { get; set; }

    /// <summary>
    ///     Gets or sets the IP address of the device.
    /// </summary>
    [MaxLength(255)]
    public required string Ip { get; set; }

    /// <summary>
    ///     Gets or sets the MAC address of the device.
    /// </summary>
    [MaxLength(255)]
    public required string Mac { get; set; }

    /// <summary>
    ///     Gets or sets a value indicating whether the device is currently connected.
    /// </summary>
    public required bool Connected { get; set; }

    /// <summary>
    ///     Navigation property to the collection of
    ///     <see cref="DeviceScriptsMapping" />s.
    ///     Represents the scripts that are mapped to this device.
    /// </summary>
    public virtual ICollection<DeviceScriptsMapping>? DeviceScriptsMappings { get; set; }
}

File: Core/src/device.py
# device.py
class Device:
    def __init__(self, name, ip, mac, connected=False, scripts=None):
        self.name = name
        self.ip = ip
        self.mac = mac
        self.scripts = scripts if scripts is not None else []
        self.connected = connected  

    def add_script(self, script):
        self.scripts.append(script)

    def to_dict(self):
        """Convert the Device object to a dictionary."""
        return {
            'name': self.name,
            'ip': self.ip,
            'mac': self.mac,
            'scripts': self.scripts,
            'connected': self.connected
        }

    def __str__(self):
        return f"Device(name={self.name}, ip={self.ip}, mac={self.mac}, scripts={self.scripts}, connected={self.connected})"
